### ISSUE TITLE
Error While using FastAPI response_model validation and filtering

### ISSUE THREAD

**Issue by jmguillens** [pydantic/pydantic#7237]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

While using FastAPI response_model validation and filtering,
I receive a ResponseValidationError(
fastapi.exceptions.ResponseValidationError
and the error disapers when eliminating the response_model from Pydantic

When I run the me endpoint everything works OK if the response_model is removed, but the object I return includes password and other stuff that I want to filter with response_model=
Once I keep response_model then it fails with all options that I have commented in the code.

Any help would be welcomed

### Example Code

```Python
from sqlalchemy import Column, Integer, String, Date, Boolean, ForeignKey
from sqlalchemy.sql.sqltypes import TIMESTAMP
from sqlalchemy.sql.expression import text


from pydantic import BaseModel, EmailStr, ConfigDict

from fastapi import APIRouter, Depends, HTTPException, status
from fastapi.security.oauth2 import OAuth2PasswordRequestForm
from sqlalchemy.orm import Session

from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

from .config import settings

SQLALCHEMY_DATABASE_URL = f"postgresql+psycopg://{settings.database_username}:{settings.database_password}@{settings.database_hostname}:{settings.database_port}/{settings.database_name}"

engine = create_engine(SQLALCHEMY_DATABASE_URL)

SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)

Base = declarative_base()


def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

class User(Base):
    __tablename__ = "users"

    id = Column(Integer, primary_key=True, nullable=False)
    username = Column(String(80), nullable=False, unique=True)
    password = Column(String(80), nullable=False)
    name = Column(String(60), nullable=False)
    surname = Column(String(80), nullable=False)
    birthday = Column(Date(), nullable=True)
    sex = Column(Integer, nullable=False, default=0)
    timezonelabel = Column(String(50), nullable=False, default="")
    istimezonelabeluserdefined = Column(
        Boolean(), nullable=False, default=False)
    created_at = Column(TIMESTAMP(timezone=True),
                        nullable=False, server_default=text('now()'))
    locale = Column(String(10), nullable=False, default="en-US")
    updated_at = Column(TIMESTAMP(timezone=True),
                        nullable=False, default=text('now()'), onupdate=text('now()'))

class MeUser(BaseModel):
    # model_config = ConfigDict(from_attributes=True)
    id: int
    username: str
    name: str
    surname: str
    birthday: str
    sex: int
    timezonelabel: str
    istimezonelabeluserdefined: bool
    locale: str

    class ConfigDict:
        from_attributes = True
    # class Config:
    #     orm_mode = True

@router.get("/me",
            status_code=status.HTTP_202_ACCEPTED,
            response_model=schemas.MeUser)
# )
async def me(db: Session = Depends(get_db),
             current_user: schemas.TokenData = Depends(oauth2.get_current_user)):
    me = db.query(models.User).filter(
        models.User.id == current_user.id).first()
    if not me:
        raise HTTPException(
            status_code=status.HTTP_403_FORBIDDEN, detail=f"Invalid Credentials")
    return me
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.0.3
        pydantic-core version: 2.3.0 release build profile
                 install path: /Users/jmg/Documents/StudeamFastApi/venv/lib/python3.11/site-packages/pydantic
               python version: 3.11.4 (v3.11.4:d2340ef257, Jun  6 2023, 19:15:51) [Clang 13.0.0 (clang-1300.0.29.30)]
                     platform: macOS-13.4.1-arm64-arm-64bit
     optional deps. installed: ['email-validator', 'typing-extensions']
```

Selected Assignee: @samuelcolvin

**Comment by JavierSanchezCastro** [comment]
Hi!

Please, provide a basic example to reproduce your error, it's hard to follow it

**Comment by samuelcolvin** [comment]
For a start, you're using `ConfigDict` wrong, please consult the docs.

Beyond that:
1. This is a question not a bug report.
2. You need to give a MRE for us to be able to help you.
